### PR TITLE
fix `Incorrect overriding of the NodeList collection` (close #1141, close #1172)

### LINKED ISSUE #1141: Incorrect overriding of the NodeList collection
The script below works fine without our proxy, but it occurs the `Uncaught TypeError: scripts[prop].getAttribute is not a function` error when we run it through proxy.
```javascript
var scripts = document.querySelectorAll('script');

for (var prop in scripts) {
    if (scripts.hasOwnProperty(prop))
        scripts[prop].getAttribute('src');
}
```
We will see the following discrepancy if we call the `Object.keys` function on our collection.
Without proxy `["0", "1", "2"]`
With proxy `["0", "1", "2", "item", "namedItem"]`

### LINKED ISSUE #1172: "getElementsByTagName" method returns collection with extra properties
Based on 
https://testcafe-discuss.devexpress.com/t/uncaught-typeerror-elm-getattribute-is-not-a-function/304

There are two extra methods `item` and `namedItem` in the `for ... in` loop.

Check all errors on this site.

### PR DESCRIPTION
**Pull request by LavrovArtem**
/cc @miherlosev @churkin

### CODE REVIEW

File: test/client/fixtures/sandbox/shadow-ui-test.js [hunk 1]
```diff
@@ -809,3 +809,16 @@ test('querySelector call should return a first non-shadowUI element (GH-1131)',
     document.head.removeChild(meta);
     document.head.removeChild(shadowMeta);
 });
+
+test('collection gotten via the querySelectorAll function should not contains enumerable item and namedItem properties (GH-1141)', function () {
+    var collection     = document.querySelectorAll('div');
+    var collectionKeys = Object.keys(collection);
```
Comment (miherlosev) [inline test/client/fixtures/sandbox/shadow-ui-test.js]: Lets' compare with a native behaviour.
```js
var nativeCollection = ....

strictEqual(nativeCollection.hasOwnPropery('name', collection.hasOwnProperty('name'), 'name')
...
var nativeNamePropDescriptop = Object.getOwnPropertyDescriptor(nativeCollection, 'name');
var namePropDescriptor = ....

strictEqual(nativeNamePropDescriptop.enumerable, namePropDescriptor.enumerable)
```

**Comment by LavrovArtem** [comment]
@testcafe-build-bot \retest

**Comment by LavrovArtem** [comment]
@testcafe-build-bot \retest

File: test/client/fixtures/sandbox/shadow-ui-test.js [hunk 2]
```diff
@@ -809,3 +809,16 @@ test('querySelector call should return a first non-shadowUI element (GH-1131)',
     document.head.removeChild(meta);
     document.head.removeChild(shadowMeta);
 });
+
+test('processed nodeList should have non-enumerable item and namedItem properties (GH-1141)', function () {
```
Comment (miherlosev) [inline test/client/fixtures/sandbox/shadow-ui-test.js]: Wrap `name` and `nameItem` to single quotes

**Comment by LavrovArtem** [comment]
FPR

File: test/client/fixtures/sandbox/shadow-ui-test.js (hunk 2 above)
Comment (LavrovArtem) [inline test/client/fixtures/sandbox/shadow-ui-test.js]: The name of test has single quotes. Maybe do you mean wrap item and namedItem to double quote?

Comment (miherlosev) [inline test/client/fixtures/sandbox/shadow-ui-test.js]: ok

**Comment by LavrovArtem** [comment]
FPR

**Comment by LavrovArtem** [comment]
ping @churkin

**Pull request merged**